test-framework/kubernetes-client/src/main/java/io/quarkus/test/kubernetes/client/WithPortForwarding.java

```diff
@@ -0,0 +1,52 @@
+package io.quarkus.test.kubernetes.client;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+import io.quarkus.test.common.QuarkusTestResource;
+
+@QuarkusTestResource(value = PortForwardingTestResource.class, restrictToAnnotatedClass = true)
+@Retention(RetentionPolicy.RUNTIME)
+@Target(ElementType.TYPE)
+public @interface WithPortForwarding {
```

Comment:
If get this right, you are using:
- pod name
- label selectors
- field selector 

for portforwading.

The thing is that usually pod name is not known in advanced and selectors are two hard to use. I would like to suggest using the `Service` name instead. It has a predicatable name (in most cases equal to the app name) and encapsulates the selector logic inside it. Most importantly, you can directly use it to portforward.

Reply:
There might not be a `Service`, though…

Reply:
Re-worked the pod identifying mechanism to make it simpler to use in simple cases while still allowing for the full selector mechanism.

Reply:
There are so many things the user need to add, including index, names that are machine generated and what not, that it's easier to require a `Service`. Besides the `quarkus-kubernetes` extension will always generate a Service as long as there is even a single port exposed.

I understand that in your primary use case there might not be a `Service` but in most cases there will be. 
If anything you should allow the use of `Service` and allow users to select.